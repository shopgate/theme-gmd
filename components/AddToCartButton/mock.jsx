/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
