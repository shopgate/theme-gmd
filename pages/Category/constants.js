/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const GRID_VIEW = 'grid';
export const LIST_VIEW = 'list';

export const SET_CATEGORY_VIEW_MODE = 'SET_CATEGORY_VIEW_MODE';
