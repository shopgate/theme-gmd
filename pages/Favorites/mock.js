/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

