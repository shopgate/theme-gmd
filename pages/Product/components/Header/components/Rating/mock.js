/*
 *  Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export {
  getElementById,
};
