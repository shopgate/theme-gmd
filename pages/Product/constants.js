/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Action Types
 */
export const OPEN_PRODUCT_GALLERY = 'OPEN_PRODUCT_GALLERY';
export const REQUEST_PRODUCT_DATA = 'REQUEST_PRODUCT_DATA';

/**
 * Max number of reviews shown
 * @type {number}
 */
export const REVIEW_PREVIEW_COUNT = 2;
