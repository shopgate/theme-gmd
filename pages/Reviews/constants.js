/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Number of reviews items that should be loaded per bundle (page or load more).
 * @type {number}
 */
export const REVIEW_ITEMS_PER_PAGE = 10;
