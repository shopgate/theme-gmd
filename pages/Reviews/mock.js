/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
