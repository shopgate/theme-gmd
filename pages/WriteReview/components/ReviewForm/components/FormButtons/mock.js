/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
};
