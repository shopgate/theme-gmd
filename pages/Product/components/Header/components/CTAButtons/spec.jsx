import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { ADD_PRODUCTS_TO_CART } from '@shopgate/pwa-common-commerce/cart/constants';
import { mockedPipelineRequestFactory } from '@shopgate/pwa-core/classes/PipelineRequest/mock';
// Import from context since it'll be mocked later
import { defaultContext } from './../../../../context';
import {
  mockedState,
  mockedVariantState,
} from './mock';

const mockedStore = configureStore([thunk]);
const mockedResolver = jest.fn();
jest.mock('@shopgate/pwa-core/classes/PipelineRequest', () => mockedPipelineRequestFactory((mockInstance, resolve, reject) => {
  mockedResolver(mockInstance, resolve, reject);
}));

jest.mock('./../../../../context');

describe('CTAs (product header)', () => {
  let store;

  afterEach(() => {
    jest.resetAllMocks();
  });
  /**
   * Creates component
   * @param {Object} state Mocked state
   * @param {Object} props Mocked props
   * @return {ReactWrapper}
   */
  const createComponent = (state, props) => {
    store = mockedStore(state);
    /* eslint-disable global-require */
    const CTAButtons = require('./index').default;
    /* eslint-enable global-require */
    return mount(<Provider store={store}><CTAButtons {...props} /></Provider>);
  };

  it('should render CTAButtons when data is available', () => {
    const productId = 'product_1';
    defaultContext.productId = productId;
    const component = createComponent(mockedState, { productId });
    expect(component).toMatchSnapshot();
    expect(component.find('FavoritesButton').exists()).toBe(true);
    expect(component.find('CartButton').exists()).toBe(true);
  });

  it('should handle add to cart', (done) => {
    const productId = 'product_2';
    defaultContext.productId = productId;
    const component = createComponent(mockedVariantState, { productId });
    component.find('CartButton').find('button').simulate('click');
    component.update();

    window.setTimeout(() => {
      const actions = store.getActions();
      expect(actions[0].type).toBe(ADD_PRODUCTS_TO_CART);
      done();
    }, 0);
  });
});
