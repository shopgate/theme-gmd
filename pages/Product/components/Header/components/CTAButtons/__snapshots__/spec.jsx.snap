// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAs (product header) should render CTAButtons when data is available 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CTAButtons)
    productId="product_1"
  >
    <CTAButtons
      dispatch={[Function]}
      isFavorite={true}
      productId="product_1"
    >
      <div
        className="css-7azkrx"
      >
        <Connect(FavoritesButton)
          active={true}
          className="css-13l587d"
          productId="product_1"
          rippleClassName="css-1tdptiz"
        >
          <FavoritesButton
            active={true}
            addFavorites={[Function]}
            className="css-13l587d"
            noShadow={false}
            onRippleComplete={[Function]}
            once={false}
            productId="product_1"
            removeFavorites={[Function]}
            removeThrottle={0}
            removeWithRelatives={false}
            rippleClassName="css-1tdptiz"
          />
        </Connect(FavoritesButton)>
        <Component>
          <Consumer>
            <Connect(CartButton)
              conditioner={
                Conditioner {
                  "conditioners": Map {},
                }
              }
              options={Object {}}
              productId="product_1"
            >
              <CartButton
                addToCart={[Function]}
                conditioner={
                  Conditioner {
                    "conditioners": Map {},
                  }
                }
                disabled={true}
                loading={false}
                options={Object {}}
                productId="product_1"
              >
                <FloatingActionButton
                  background="#ccc"
                  className="css-mp6sl7"
                  onClick={[Function]}
                  raised={true}
                  ref={null}
                  size="big"
                  type="button"
                >
                  <button
                    className="css-1dbx872 css-mp6sl7 css-1kx79ob css-1g5n8t6"
                    onClick={[Function]}
                    style={
                      Object {
                        "background": "#ccc",
                      }
                    }
                    type="button"
                  >
                    <CartButtonIcon
                      onSuccess={[Function]}
                      success={false}
                    >
                      <Transition
                        appear={false}
                        enter={true}
                        exit={true}
                        in={false}
                        mountOnEnter={false}
                        onEnter={[Function]}
                        onEntered={[Function]}
                        onEntering={[Function]}
                        onExit={[Function]}
                        onExited={[Function]}
                        onExiting={[Function]}
                        timeout={800}
                        unmountOnExit={false}
                      >
                        <div
                          className={
                            Object {
                              "data-css-1h3cr19": "",
                            }
                          }
                        >
                          <CartPlus
                            className="css-b18rpa"
                            size={24}
                          >
                            <Icon
                              className="css-b18rpa"
                              color={null}
                              content="<path d=\\"M0 0h24v24H0zm18.31 6l-2.76 5z\\" fill=\\"none\\"/><path d=\\"M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z\\"/>"
                              size={24}
                              viewBox="0 0 24 24"
                            >
                              <svg
                                className="css-yys9hb css-b18rpa"
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "<path d=\\"M0 0h24v24H0zm18.31 6l-2.76 5z\\" fill=\\"none\\"/><path d=\\"M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z\\"/>",
                                  }
                                }
                                style={
                                  Object {
                                    "fill": null,
                                    "fontSize": 24,
                                  }
                                }
                                viewBox="0 0 24 24"
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </Icon>
                          </CartPlus>
                          <Tick
                            className="css-v74gh2"
                            size={24}
                          >
                            <Icon
                              className="css-v74gh2"
                              color={null}
                              content="<path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"/>"
                              size={24}
                              viewBox="0 0 24 24"
                            >
                              <svg
                                className="css-yys9hb css-v74gh2"
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "<path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"/>",
                                  }
                                }
                                style={
                                  Object {
                                    "fill": null,
                                    "fontSize": 24,
                                  }
                                }
                                viewBox="0 0 24 24"
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </Icon>
                          </Tick>
                        </div>
                      </Transition>
                    </CartButtonIcon>
                  </button>
                </FloatingActionButton>
              </CartButton>
            </Connect(CartButton)>
          </Consumer>
        </Component>
      </div>
    </CTAButtons>
  </Connect(CTAButtons)>
</Provider>
`;
